tests/parametric/test_parametric_endpoints.py

```diff
@@ -0,0 +1,733 @@
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+        assert span["meta"]["key"] == "value"
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDSpan_Set_Metric:
+    def test_set_metric(self, test_agent, test_library):
+        """
+        Validates that /trace/span/set_meta sets a metric key value pair on a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - key: str
+        - value: Union[int, float]
+        Supported Return Values:
+        """
+
+        with test_library:
+            with test_library.start_span("span_meta") as s1:
+                s1.set_metric("key", 1)
+
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+        assert span["metrics"]["key"] == 1
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDSpan_Set_Error:
+    def test_set_error(self, test_agent, test_library):
+        """
+        Validates that /trace/span/set_error sets an error on a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - type: str
+        - message: str
+        - stack: str
+        Supported Return Values:
+        """
+
+        with test_library:
+            with test_library.start_span("span_set_error") as s1:
+                s1.set_error("MyException", "Parametric tests rock", "fake_stacktrace")
+
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+        assert span["error"] == 1
+        assert span["meta"]["error.type"] == "MyException"
+        assert span["meta"].get("error.message", span["meta"].get("error.msg")) == "Parametric tests rock"
+        assert span["meta"]["error.stack"] == "fake_stacktrace"
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDSpan_Set_Resource:
+    def test_set_resource(self, test_agent, test_library):
+        """
+        Validates that /trace/span/set_resource sets a resource name on a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - resource: str
+        Supported Return Values:
+        """
+        with test_library:
+            with test_library.start_span("span_set_resource", "old_resource") as s1:
+                s1.set_resource("new_resource")
+
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+        assert span["resource"] == "new_resource"
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDSpan_Add_Link:
+    def test_add_link(self, test_agent, test_library):
+        """
+        Validates that /trace/span/add_link adds a link to a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - parent_id: Union[int, str]
```

Comment:
```suggestion
```

Reply:
These arguments allow us to do the the following operation in a parametric endpoint:
SpanA.add_link(SpanB, attributes)` where `SpanA.span_id==span_id and SpanB.span_id==parent_id`.

We should honestly rename `parent_id` but this is the spec we currently have 😭 . We can fix it in a follow up PR.